docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,129 @@
+        val rs1Out     =  Input(Bits(dl_size.W))
+        val pcOut      =  Input(Bits(dl_size.W))
+        val aluMux1Out = Output(Bits(dl_size.W))
+    })
+
+    // Default value for aluMux1Out
+    io.aluMux1Out := 0.U
+
+    switch (io.aluMux1Sel) {
+        is (rs1out) {
+            io.aluMux1Out  := io.rs1Out
+        }
+        is (pcout) {
+            io.aluMux1Out  := io.pcOut
+        }
+    }
+}
+```
+```scala mdoc:verilog
+import 
+ChiselStage.emitVerilog.emitVerilog(new AluMux1File(dl_size = 32) )
+```
+
+
+ChiselEnum also allows for the user to define variables by passing in the value shown below. Note that the value must be increasing.
+
+```scala mdoc:silent
+object Opcode extends ChiselEnum {
+    val load  = Value(0x03.U) // i load  -> 000_0011
+    val imm   = Value(0x13.U) // i imm   -> 001_0011
+    val auipc = Value(0x17.U) // u auipc -> 001_0111
+    val store = Value(0x23.U) // s store -> 010_0011
+    val reg   = Value(0x33.U) // r reg   -> 011_0011
+    val lui   = Value(0x37.U) // u lui   -> 011_0111
+    val br    = Value(0x63.U) // b br    -> 110_0011
+    val jalr  = Value(0x67.U) // i jalr  -> 110_0111
+    val jal   = Value(0x6F.U) // j jal   -> 110_1111
+}
+```
+
+The user can 'jump' to a value and continue incrementing by passing a start point then using a regular Value assignment. 
+
+```scala mdoc:silent
+object BranchFunct3 extends ChiselEnum {
+    val beq, bne = Value
+    val blt = Value(4.U)
+    val bge, bltu, bgeu = Value
+    /**
+        "beq"  -> 0.U,
+        "bne"  -> 1.U,
+        "blt"  -> 4.U,
+        "bge"  -> 5.U,
+        "bltu" -> 6.U,
+        "bgeu" -> 7.U
+    */
+}
+```
+
+## Testing
+
+When testing your modules, the `.Type` and `.litValue` attributes allow for the the objects to be passed as parameters and for the value to be converted to BigInt type. Note that BigInts cannot be casted to Int with `.asInstanceOf[Int]`, they use their own methods like `toInt`. Please review the [scala.math.BigInt](https://www.scala-lang.org/api/2.12.5/scala/math/BigInt.html) page for more details!
+
+```scala mdoc:silent
+def expectedSel(sel: AluMux1Sel.Type): Boolean = sel match {
+  case AluMux1Sel.rs1out => (sel.litValue == 0)
+  case AluMux1Sel.pcout  => (sel.litValue == 1)
+  case _                 => false
+}
+```
+
+The ChiselEnum type also has methods `.all` and `.getWidth` where all returns all of the enum instances and getWidth returns the width of the hardware type.
+
+## Workarounds
+
+As of 2/26/2021, the width of the values is always infered so to get around this, just add an extra state that forces the width that is desired. 
+
+```scala mdoc:silent
+object StoreFunct3 extends ChiselEnum {
+    val sb, sh, sw = Value
+    val ukn = Value(7.U)
+    /**
+        "sb" -> 0.U,
+        "sh" -> 1.U,
+        "sw" -> 2.U
+    */
+}
+```
+
+Additionally I don't believe that signed values are supported so if you want the value signed, you must cast the UInt with `.asSInt`.
```

Comment:
Before this PR is merged let's get rid of the `I don't believe` and just make sure what is written is correct.